File: salomon/tools/WinSalomon/trunk/WinSalomon/Project.cpp
/*
 * Copyright (C) 2005 Salomon Team
 *
 * This file is part of Salomon.
 *
 * Salomon is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 2.1 of the License, or (at your option) any later version.
 *
 * Salomon is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public
 * License along with Salomon; if not, write to the Free Software
 * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
 * 
 */


#include "project.h"
#include "stringhelper.h"

const char* Project::CLASS_NAME = "salomon/engine/project/IProject";

DLL_SHARE std::string Project::getInfo()
{
	return NULL;
}

DLL_SHARE std::string Project::getName()
{
	return NULL;
}

DLL_SHARE int Project::getProjectID()
{
	return NULL;
}

DLL_SHARE void Project::setInfo(std::string& info)
{
	std::cout << "setInfo...";

	jmethodID setInfoMethod = this->findMethod("setInfo", "(Ljava/lang/String;)V");	

	jstring strInfo = StringHelper::getString(getEnv(), info.c_str());
	
	this->getEnv()->CallVoidMethod(this->getObject(), setInfoMethod, strInfo);
}

DLL_SHARE void Project::setName(std::string& name)
{
	std::cout << "setName...";

	jmethodID setNameMethod = this->findMethod("setName", "(Ljava/lang/String;)V");	

	jstring strName = StringHelper::getString(getEnv(), name.c_str());
	
	this->getEnv()->CallVoidMethod(this->getObject(), setNameMethod, strName);
}

DLL_SHARE void Project::setProjectID(int projectId)
{
}

DLL_SHARE TaskManager* Project::getTaskManager()
{
	std::cout << "Getting TaskManager...";

	jmethodID getTaskManagerMethod = this->findMethod("getTasksManager", "()Lsalomon/engine/task/ITaskManager;");

	jobject taskManager = this->getEnv()->CallObjectMethod(this->getObject(), getTaskManagerMethod);

	TaskManager* result = new TaskManager(getEnv(), taskManager);

	if (result != 0)
	{
		std::cout << "success" << std::endl;
	}
	else
	{
		std::cout << "failure" << std::endl;
	}

	return result;
}

File: src/salomon/core/project/Project.java

package salomon.core.project;

/**
 * Represents a project, it is an implementation of IProject interface.
 *  
 */
final class Project implements IProject
{
	private String _info = null;

	private String _name = null;

	private int _projectID = 0;

	public Project()
	{
	}

	/**
	 * @return Returns the info.
	 */
	public String getInfo()
	{
		return _info;
	}

	/**
	 * @return Returns the name.
	 */
	public String getName()
	{
		return _name;
	}

	/**
	 * @return Returns the projectID.
	 */
	public int getProjectID()
	{
		return _projectID;
	}

	/**
	 * @param info The info to set.
	 */
	public void setInfo(String info)
	{
		_info = info;
	}

	/**
	 * @param name The name to set.
	 */
	public void setName(String name)
	{
		_name = name;
	}

	/**
	 * @param projectId The projectID to set.
	 */
	public void setProjectID(int projectId)
	{
		_projectID = projectId;
	}

	public String toString()
	{
		return "[" + _projectID + ", " + _name + ", " + _info + "]";
	}
}